### PR TITLE
Mark transaction as `internal_error` in case of unhandled errors

### PR DESCRIPTION
**Pull request by marandaneto**
## :bulb: Motivation and Context
Closes https://github.com/getsentry/sentry-dart/issues/1182

## :pencil: Checklist

- [X] I reviewed submitted code
- [X] I added tests to verify changes
- [ ] I updated the docs if needed
- [X] All tests passing
- [ ] No breaking changes

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,27 +8,10 @@
   - captureFailedRequests
   - enableStructuredDataTracing
   - enableUserInteractionTracing
+- Mark transaction as internal_error in case of unhandled errors ([#1218](https://github.com/getsentry/sentry-dart/pull/1218))
 
 ## 7.0.0-alpha.1
 
-### Various fixes & improvements
```
Comment (marandaneto) [inline CHANGELOG.md:14]: Just fixing the changelog.

File: dart/lib/src/isolate_error_integration.dart [hunk 2]
```diff
@@ -27,54 +27,61 @@ class IsolateErrorIntegration extends Integration {
...
 
-RawReceivePort _createPort(Hub hub, SentryOptions options) {
-  return RawReceivePort(
-    (dynamic error) async {
-      await handleIsolateError(hub, options, error);
-    },
-  );
-}
+  /// Parse and raise an event out of the Isolate error.
+  @visibleForTesting
```
Comment (marandaneto) [inline dart/lib/src/isolate_error_integration.dart:32]: Just moved from a standalone function to inside of the class.

File: dart/lib/src/isolate_error_integration.dart [hunk 3]
```diff
@@ -27,54 +27,61 @@ class IsolateErrorIntegration extends Integration {
...
-    final event = SentryEvent(
-      throwable: throwableMechanism,
-      level: SentryLevel.fatal,
-      timestamp: hub.options.clock(),
-    );
+      // marks the span status if none to `internal_error` in case there's an
+      // unhandled error
+      hub.configureScope((scope) => {
+            scope.span?.status ??= const SpanStatus.internalError(),
+          });
```
Comment (marandaneto) [inline dart/lib/src/isolate_error_integration.dart:74]: That's the added code.

File: dart/lib/src/run_zoned_guarded_integration.dart [hunk 4]
```diff
@@ -20,6 +22,40 @@ class RunZonedGuardedIntegration extends Integration {
   /// Needed to check if we somehow caused a `print()` recursion
   bool _isPrinting = false;
 
+  @visibleForTesting
```
Comment (marandaneto) [inline dart/lib/src/run_zoned_guarded_integration.dart:25]: Just moved to the class, to be tested.

File: dart/lib/src/run_zoned_guarded_integration.dart [hunk 5]
```diff
@@ -20,6 +22,40 @@ class RunZonedGuardedIntegration extends Integration {
...
+      throwable: throwableMechanism,
+      level: SentryLevel.fatal,
+      timestamp: hub.options.clock(),
+    );
+
+    // marks the span status if none to `internal_error` in case there's an
+    // unhandled error
+    hub.configureScope((scope) => {
+          scope.span?.status ??= const SpanStatus.internalError(),
+        });
```
Comment (marandaneto) [inline dart/lib/src/run_zoned_guarded_integration.dart:54]: Added code.

File: flutter/test/integrations/on_error_integration_test.dart [hunk 6]
```diff
@@ -9,134 +9,170 @@ import '../mocks.mocks.dart';
 import 'mock_platform_dispatcher.dart';
 
 void main() {
-  TestWidgetsFlutterBinding.ensureInitialized();
+  group(OnErrorIntegration, () {
```
Comment (marandaneto) [inline flutter/test/integrations/on_error_integration_test.dart:12]: Just moved within the group

File: flutter/test/integrations/on_error_integration_test.dart [hunk 7]
```diff
@@ -9,134 +9,170 @@ import '../mocks.mocks.dart';
...
-
-    final event = verify(
-      await fixture.hub
-          .captureEvent(captureAny, stackTrace: captureAnyNamed('stackTrace')),
-    ).captured.first as SentryEvent;
-
-    final throwableMechanism = event.throwableMechanism as ThrowableMechanism;
-    expect(throwableMechanism.mechanism.handled, true);
-  });
+    test('marks transaction as internal error if no status', () async {
```
Comment (marandaneto) [inline flutter/test/integrations/on_error_integration_test.dart:144]: Added test

**Review by brustolin — APPROVED** [review]
LGTM

**Pull request merged**